src/server/routes/editor.js

```diff
@@ -174,6 +174,26 @@ router.post('/edit/handler', auth.isAuthenticatedForHandler, (req, res) => {
 	handler.sendPromiseResult(res, runAsync(), search.indexEntity);
 });
 
+router.post('/privs/edit/handler', auth.isAuthenticatedForHandler, async (req, res, next) => {
+	const {Editor} = req.app.locals.orm;
+	try {
+		const editor = await Editor
+			.forge({id: parseInt(req.body.targetUserId, 10)})
+			.fetch({require: true})
+			.catch(Editor.NotFoundError, () => next(new error.NotFoundError('Editor not found', req)));
+		if (editor.get('privs') === req.body.newPrivs) {
+			return next(new error.FormSubmissionError(
+				'No change to Privileges', req
+			));
+		}
+		const modifiedEditor = await editor.save({privs: req.body.newPrivs});
+		return res.status(200).send();
+	}
+	catch (err) {
+		return next(err);
+	}
+});
```

Comment:
You should add a check, if the user who has sent the request has the privilege to perform this operaiton.

Reply:
We'll eventually have to add middlewares for privilege-based access on almost all the routes. I think that will cover this.